packages/typespec-ts/src/modular/serialization/util.ts

```diff
@@ -0,0 +1,90 @@
+import { ImportType } from "@azure-tools/rlc-common";
+import { UsageFlags } from "@typespec/compiler";
+import { toPascalCase } from "../../utils/casingUtils.js";
+import {
+  isSpecialHandledUnion,
+  isSpecialUnionVariant
+} from "../buildSerializeUtils.js";
+import { Property, Type } from "../modularCodeModel.js";
+
+export type Imports = Array<{
+  importType: ImportType | "serializers";
+  name: string;
+}>;
+export type ModularTypeName = string;
+export type SerializerMap = Record<ModularTypeName, SerializerMetadata>;
+export interface SerializerMetadata {
+  rlcTypeName: string;
+  rlcTypeAlias: string;
+  type: Type;
+  serializerFunctionName?: string;
+  deserializerFunctionName?: string;
+}
+export type SerializerOutput = { expr: string; imports?: Imports };
+
+export function getAllNamedAncestors(t: Type): Type[] {
+  const map = new Map<string, Type>();
+  const stack: Type[] = [t];
+  while (stack.length) {
+    const type = stack.pop()!;
+    if (type.name && !map.has(type.name)) {
+      map.set(type.name, type);
+      stack.push(...(t.parents ?? []));
+    }
+  }
+  return Array.from(map.values());
+}
+
+export function getEncodingFormat(type: { format?: string }) {
+  const supportedFormats = ["base64url", "base64", "byte"];
```

Comment:
https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-util/src/bytesEncoding.ts#L5 I think that's core supported format ?

Reply:
This function has already been checked in elsewhere in the codebase as-is. I think UTF-8 is missing because tsp doesn't support encoding string data as base64. I don't find the docs to be clear about this though.

Reply:
tsp `@encode` decorators accepts open string, so if we have something like 
```typespec
model Widget {
  @key id: string;
  weight: int32;
  color: "red" | "blue";
  @encode("utf-8")
  prop: string;
}
```
it won't report any errors. Though I do know typespec has a BytesKnownEncoding only specifies base64 and base64url https://github.com/microsoft/typespec/blob/6f68753b191fd5171eb5153fb7b8a547b9942b7f/packages/compiler/src/lib/decorators.ts#L693 

And I did ask them to add some common used type into the BytesKnownEncoding, but they refused to https://github.com/microsoft/typespec/issues/2204  😢